esphome/components/template/text/__init__.py

```diff
@@ -24,11 +24,11 @@
 
 def validate(config):
     if CONF_LAMBDA in config:
-        if config[CONF_OPTIMISTIC]:
+        if config[CONF_OPTIMISTIC] and config[CONF_OPTIMISTIC]:
```

Comment:
No point in repeating the same test:
```suggestion
        if config[CONF_OPTIMISTIC]:
```